shared/p2p/service.go

```diff
@@ -145,7 +145,7 @@ func (s *Server) emit(feed *event.Feed, msg *floodsub.Message, msgType reflect.T
 		return
 	}
 
-	i := feed.Send(Message{Data: d})
+	i := feed.Send(Message{Ctx: ctx, Data: d})
```

Comment:
I think it is unintentional that we are creating another message. Can you pass the message as an argument and assign the data? 

Otherwise, we'll lose the peer data (which is supposed to be assigned, in a future PR with single peer communication).